zebra-state/src/service.rs

```diff
@@ -508,23 +495,21 @@ impl StateService {
     }
 
     /// Return the [`transparent::Utxo`] pointed to by `outpoint`, if it exists
-    /// in any chain, or in any pending block.
+    /// in any non-finalized chain, or in any pending block.
     ///
     /// Some of the returned UTXOs may be invalid, because:
+    /// - they have already been spent,
     /// - they are not in the best chain, or
     /// - their block fails contextual validation.
-    pub fn any_utxo(&self, outpoint: &transparent::OutPoint) -> Option<transparent::Utxo> {
-        // We ignore any UTXOs in FinalizedState.queued_by_prev_hash,
-        // because it is only used during checkpoint verification.
+    ///
+    /// Finalized UTXOs are handled via [`ReadRequest::ChainUtxo`].
+    pub fn any_non_finalized_utxo(
+        &self,
+        outpoint: &transparent::OutPoint,
+    ) -> Option<transparent::Utxo> {
         self.mem
```

Comment:
Move to using the `chain` from the `ReadStateService`

Reply:
Fixed in ecb0259246 by sending the whole `NonFinalizedState` to the `ReadStateService`.